src/chain.h

```diff
@@ -63,6 +67,12 @@ class CBlockFileInfo
          nBlocks = 0;
          nSize = 0;
          nUndoSize = 0;
+         nFlags = 0;
```

Comment:
These changes belong in the `CBlockIndex` class, not here in `CBlockFileInfo`.

Also, `CBlockIndex` is completely missing the `SetNull` method, which is called on `COutPoint prevoutStake` across the file (and possibly elsewhere)

Take a look at the file on 0.17 here:
https://github.com/CloakProject/codename-phoenix/blob/0.17.cloak/src/chain.h

Please make sure you're careful when you move code to appropriate locations - this is important as some properties are purposely mem-only or not serialized at all depending on usage.